toolkit/tools/pkg/imagecustomizerlib/customizepackages.go

```diff
@@ -38,10 +40,18 @@ var (
 )
 
 func addRemoveAndUpdatePackages(buildDir string, baseConfigPath string, config *imagecustomizerapi.OS,
-	imageChroot *safechroot.Chroot, rpmsSources []string, useBaseImageRpmRepos bool,
+	imageChroot *safechroot.Chroot, rpmsSources []string, useBaseImageRpmRepos bool, snapshotTime string,
 ) error {
 	var err error
 
+	if snapshotTime != "" {
+		err = createTempTdnfConfigWithSnapshot(imageChroot, imagecustomizerapi.PackageSnapshotTime(snapshotTime))
+		if err != nil {
+			return err
+		}
+		defer cleanupSnapshotTimeConfig(imageChroot)
```

Comment:
If `cleanupSnapshotTimeConfig` fails, then the customization process should fail.